lib/usb/sys/xpad.c

```diff
@@ -72,8 +74,8 @@ static void xpad_irq(struct urb *urb, struct pt_regs *regs)
 
 static int GetPadNum(const char* szDevPath)
 {
-	int pos			= strlen(szDevPath) - 3;
-	int portchar	= szDevPath[pos];
+	int pos = (XboxHardwareInfo.Flags & XBOX_HW_FLAG_INTERNAL_USB_HUB) ? 2 : 0;
```

Comment:
This should be in common USB code; either as a function, or as modification to all incoming device paths (which might prevent detection of non-connected USB ports in Xbox 1.0; a very niche use-case).